### ISSUE TITLE
[Dropdown]: if titleText isn't provided DAP violation  

### ISSUE THREAD

**Issue by abbeyhrt** [carbon-design-system/carbon#6391]
## Detailed description

With the recent changes to Dropdown, if titleText isn't defined, there is a DAP violation. It's possible that this would also impact Multiselect but I didn't check. 

> What version of the Carbon Design System are you using?

v10.15 and v7.15

> What did you expect to happen?

No DAP violation since titleText isn't a required prop. 

> What happened instead?

There's a DAP violation if titleText isn't defined

![Screen Shot 2020-07-06 at 10 33 40 AM](https://user-images.githubusercontent.com/17281178/86611491-574d0b80-bf74-11ea-9721-c3053f989354.png)

> Please create a reduced test case in CodeSandbox

https://codesandbox.io/s/beautiful-paper-o26x4

**Comment by jhpedemonte** [comment]
Yes, this does impact MultiSelect.

The issue is that the `<label>` element isn't rendered if `titleText` isn't provided. However, other elements still provide the `aria-labelledby` pointing to the (non-existent) `<label>`.

A couple of options:
1. Make `titleText` required and always render `<label>`, but allow hiding (`aria-labelledby` can point to hidden elements).
2. If `titleText` isn't set, don't add `aria-labelledby` to other elements. Not sure what impact this has on accessibility of fully rendered component, in this case.

**Comment by dakahn** [comment]
Not having a visible label is a violation of [WCAG Success Criteria 3.3.2](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions). Even if the label is programmatically accessible to users of assistive technologies we still need a form control/input to be visually labeled to meet A Level requirements. So while this isn't the ideal error -- it does call out "hey, your input/control has no label", which is a violation.

**Comment by abbeyhrt** [comment]
That makes sense! We are not labelling it visually haha so definitely a violation there. It sucks that `titleText` can't be changed to be required but I get that it's a breaking change, oh well! Feel free to close!

**Comment by jhpedemonte** [comment]
I have a couple of scenarios:
1. A MultiSelect where I provide the label and don't want to use the built-in label.
2. MultiSelects used within a table, where the header describes the selects and I still don't want to show the built-in label.

Carbon does not currently allow pointing to a different label (outside of the component) for the `aria-labelledby` value. As a workaround, for both cases, this can be handled by supplying the same label text as `titleText` but hiding it. 

> So while this isn't the ideal error -- it does call out "hey, your input/control has no label", which is a violation.

But the code as it currently exists is wrong: it allows to _not_ show the label but still renders other code as if that label exists. This is wrong. If this needs to be handled for a11y, then we need to handle this some other way.

**Comment by joshblack** [comment]
Could we do something like forward `aria-labelledby` to `getButtonProps` (or whatever relevant Downshift internal getter uses `aria-labelledby` to point to the label) to support the custom label use-case?

---

Separately, definitely think we could add a warning if `titleText` is not provided, or when we can't determine if this has a visual label. In the future we could make it required as part of a breaking change if this is indeed a requirement, but we'll still need to provide a recommendation for the "action menu" use-case.

**Comment by jhpedemonte** [comment]
> Could we do something like forward `aria-labelledby` to `getButtonProps`

Do you mean that if I were to set `aria-labelledby` attribute on `MultiSelect`, it would be forwarded into the underlying elements? If so, that would work well for the scenarios I outlined above, giving the user control over this. But defaults to what is there now if attribute not provided.

**Comment by dakahn** [comment]
Closing this up! We can reopen if there's something we missed, yall :cowboy_hat_face:

**Comment by joshblack** [comment]
Re-opening so we can track adding this functionality in 👀  (specifically the `aria-labelledby`) after chatting with @dakahn!

**Comment by joshblack** [comment]
Ah, diving into this it seems like it can be tackled already with `downshiftProps` on each of the dropdown variants. If you supply [`labelId`](https://github.com/downshift-js/downshift#labelid) to `downshiftProps` it will be correctly be added to the button.

I think we'll defer to this instead of adding explicit `aria-labelledby` support for the time being 👍
